### PR TITLE
Move nightly CI tests to the main branch

### PR DESCRIPTION
**Pull request by PatKamin**
Fuzz tests set up with cron don't run because cron scheduled workflows are triggered only for the default branch.
Add nightly workflow file to the default branch with fuzz tests specified to run on 'adapters' branch.

### CODE REVIEW

**Review by pbalcer — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]
This is a little "heh", as we discussed (if we introduce any change in fuzzing tests execution on adapters branch, this workflow should be updated on a sep. PR to the main branch), but it looks clean and hopefully will work :)

Please remove after that the `nightly.yml` from the adapters branch, I guess.

**Comment by PatKamin** [comment]
> @PatKamin, actually, did you test this on your fork (e.g., on `push` trigger)?

Turned the PR to draft until I verify on my fork with cron schedule.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        -B${{github.workspace}}/build
+        -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+        -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+        -DUR_ENABLE_TRACING=ON
+        -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+        -DUR_BUILD_TESTS=ON
+        -DUR_USE_ASAN=ON
+        -DUR_USE_UBSAN=ON
+        -DUR_DPCXX=${{github.workspace}}/dpcpp_compiler/bin/clang++
+
```
Comment (lukaszstolarczuk) [inline .github/workflows/nightly.yml:45]: should we enable developer_mode here...?

Comment (PatKamin) [reply .github/workflows/nightly.yml:45]: Flags from the developer mode are not required for fuzzer to run. If there are some warnings that these flags would trigger during the build, other jobs will detect them.

**Pull request merged**